packages/hardhat/contracts/GardensTemplate.sol

```diff
@@ -19,72 +26,123 @@ contract GardensTemplate is BaseTemplate, AppIdsXDai {
+     * @param _gardenTokenName DAO governance new token name
+     * @param _gardenTokenSymbol DAO governance new token symbol
+     * @param _commonPoolAmount Initial amount of tokens in the funding pool. Unused if _existingToken is set.
+     * @param _gardenTokenLiquidity Amount of tokens to use as liquidity, paired with Honey. Unused if _existingToken is set.
      * @param _disputableVotingSettings Array of [voteDuration, voteSupportRequired, voteMinAcceptanceQuorum, voteDelegatedVotingPeriod,
      *    voteQuietEndingPeriod, voteQuietEndingExtension, voteExecutionDelay] to set up the voting app of the organization
      */
     function createDaoTxOne(
-        string _voteTokenName,
-        string _voteTokenSymbol,
-        uint256 _fundingPoolStake,
+        ERC20 _existingToken,
+        string _gardenTokenName,
+        string _gardenTokenSymbol,
+        uint256 _commonPoolAmount,
+        uint256 _gardenTokenLiquidity,
         uint64[7] _disputableVotingSettings
     ) public {
         require(_disputableVotingSettings.length == 7, ERROR_BAD_VOTE_SETTINGS);
 
         (Kernel dao, ACL acl) = _createDAO();
-        MiniMeToken voteToken = _createToken(_voteTokenName, _voteTokenSymbol, TOKEN_DECIMALS);
+        ERC20 existingToken = _existingToken; // Prevents stack too deep error
+        MiniMeToken gardenToken = _createToken(_gardenTokenName, _gardenTokenSymbol, TOKEN_DECIMALS);
+
         Agent agent = _installDefaultAgentApp(dao);
+        DisputableVoting disputableVoting = _installDisputableVotingApp(dao, gardenToken, _disputableVotingSettings);
+        IHookedTokenManager hookedTokenManager = _installHookedTokenManagerApp(dao, gardenToken, existingToken);
 
-        DisputableVoting disputableVoting = _installDisputableVotingApp(dao, voteToken, _disputableVotingSettings);
-        IHookedTokenManager hookedTokenManager = _installHookedTokenManagerApp(dao, voteToken);
+        if (existingToken == address(0)) {
+            _createPermissionForTemplate(acl, hookedTokenManager, hookedTokenManager.MINT_ROLE());
+            hookedTokenManager.mint(agent, _commonPoolAmount);
+            hookedTokenManager.mint(address(this), _gardenTokenLiquidity);
+            gardenToken.approve(address(honeyswapRouter), _gardenTokenLiquidity);
 
-        _createPermissionForTemplate(acl, hookedTokenManager, hookedTokenManager.MINT_ROLE());
-        hookedTokenManager.mint(agent, _fundingPoolStake);
+            uint256 honeyLiquidityToAdd = honeyPriceOracle.consult(stableToken, XDAI_IN_HNY_REQUIRED_FOR_NEW_TOKEN, honeyToken);
+            honeyToken.transferFrom(msg.sender, address(this), honeyLiquidityToAdd);
+            honeyToken.approve(address(honeyswapRouter), honeyLiquidityToAdd);
+
+            honeyswapRouter.addLiquidity(honeyToken, gardenToken, honeyLiquidityToAdd, _gardenTokenLiquidity, 0, 0, BURN_ADDRESS, now);
```

Comment:
I imagine you already check this. The honeyswap pair is created here if one doesn't exist?